PhysicsTools/SelectorUtils/interface/VersionedSelector.h

```diff
@@ -150,42 +155,52 @@ initialize( const edm::ParameterSet& conf ) {
       << "You have supplied a null/empty cutflow to VersionedIDSelector,"
       << " please add content to the cuflow and try again.";
   }
+  
   // this lets us keep track of cuts without knowing what they are :D
   std::vector<edm::ParameterSet>::const_iterator cbegin(cutflow.begin()),
     cend(cutflow.end());
   std::vector<edm::ParameterSet>::const_iterator icut = cbegin;
-  for( ; icut != cend; ++icut ) {   
+  std::map<std::string,unsigned> cut_counter;
+  for( ; icut != cend; ++icut ) {  
+    std::stringstream realname;
     const std::string& name = icut->getParameter<std::string>("cutName");
+    if( !cut_counter.count(name) ) cut_counter[name] = 0;      
+    realname << name << "_" << cut_counter[name];
     const bool needsContent = 
       icut->getParameter<bool>("needsAdditionalProducts");     
     const bool ignored = icut->getParameter<bool>("isIgnored");
     candf::CandidateCut* plugin = 
-#if !defined(__CINT__) && !defined(__MAKECINT__) && !defined(__REFLEX__)
       CutApplicatorFactory::get()->create(name,*icut);
-#else
-      nullptr;
-#endif
     if( plugin != nullptr ) {
-      cuts_.push_back(boost::shared_ptr<candf::CandidateCut>(plugin));
+      cuts_.push_back(std::shared_ptr<candf::CandidateCut>(plugin));
     } else {
       throw cms::Exception("BadPluginName")
 	<< "The requested cut: " << name << " is not available!";
     }
     needs_event_content_.push_back(needsContent);
-    this->push_back(name);
-    this->set(name);
-    if(ignored) this->ignoreCut(name);
+    const std::string therealname = realname.str();
+    this->push_back(therealname);
+    this->set(therealname);
+    if(ignored) this->ignoreCut(therealname);
+    cut_counter[name]++;
   }    
+
   //have to loop again to set cut indices after all are filled
   icut = cbegin;
+  cut_counter.clear();
   for( ; icut != cend; ++icut ) {
-    const std::string& name = icut->getParameter<std::string>("cutName");
-    cut_indices_.push_back(typename Selector<T>::index_type(&(this->bits_),name));
+    std::stringstream realname;
+    const std::string& name = icut->getParameter<std::string>("cutName");    
+    if( !cut_counter.count(name) ) cut_counter[name] = 0;      
+    realname << name << "_" << cut_counter[name];
+    cut_indices_.push_back(typename Selector<T>::index_type(&(this->bits_),realname.str()));    
+    cut_counter[name]++;
   }
+  
   initialized_ = true;
 }
 
-#if !defined(__CINT__) && !defined(__MAKECINT__) && !defined(__REFLEX__)
+//#if !defined(__CINT__) && !defined(__MAKECINT__) && !defined(__REFLEX__)
```

Comment:
all other cases were cleanly removed in this file; why just commenting out here?